neo/Consensus/ConsensusService.cs

```diff
@@ -379,7 +382,7 @@ private void OnPrepareRequestReceived(ConsensusPayload payload, PrepareRequest m
 
         private void OnPrepareResponseReceived(ConsensusPayload payload, PrepareResponse message)
         {
-            if (context.PreparationPayloads[payload.ValidatorIndex] != null) return;
+            if (context.PreparationPayloads[payload.ValidatorIndex] != null || context.ViewChanging()) return;
```

Comment:
You should allow receiving `PrepareRequest` and `PrepareResponse` even when `ViewChanging() == true`. Because `ViewChanging()` indicates that I want to change views, it may not be able to change successfully.

Reply:
If it accepts them it could commit in the current view, but it has already sent a change view to the next view. This will lead to nodes getting stuck if we allow it.

Reply:
Specifically it will cause them to get stuck because it could both commit in this view, and also allow other nodes to change view erroneously to the next view. However in that view they will never be able to get enough commits now to create a block.

Reply:
if i want to change my view, i can't finish the block right?

Reply:
@shargon, you can finish the block acting as a bridge (receiving commits and aggregating them) but you can not change your state to `commitment`.

Reply:
The flag used has been renamed and improvements have been made resolving this now and awaiting further review.